### PR TITLE
[READY]Refactors uplinks to a component!

### PR DESCRIPTION
**Pull request by silicons**

### CODE REVIEW

**Comment by Qustinnus** [comment]
shut up and take my GBP

**Comment by MoreRobustThanYou** [comment]
do storage next

**Comment by GuyonBroadway** [comment]
I bet uplinks are part of kevinz fetish.

**Review by Cyberboss — CHANGES_REQUESTED** [review]
This is too much of a "path change" port. Take a look at my old attempt and base it off that.#29785

File: code/datums/components/uplink.dm [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+/datum/component/uplink
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	var/name = "syndicate uplink"
+	var/active = FALSE
+	var/lockable = TRUE
+	var/locked = TRUE
+	var/telecrystals
+	var/selected_cat = null
+	var/owner = null
+	var/datum/game_mode/gamemode = null
```
Comment (Cyberboss) [inline code/datums/components/uplink.dm]: remove = nulls

File: code/datums/uplink_items.dm [hunk 2]
```diff
@@ -1,1429 +1,1396 @@
...
-				continue
-			if(U.telecrystals < I.cost)
-				continue
-			possible_items += I
-
-	if(possible_items.len)
-		var/datum/uplink_item/I = pick(possible_items)
-		U.telecrystals -= I.cost
-		U.spent_telecrystals += I.cost
+GLOBAL_LIST_EMPTY(uplink_items) // Global list so we only initialize this once.
```
Comment (Cyberboss) [inline code/datums/uplink_items.dm]: Fix line endings

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Pull request merged**